utils/data_manager/modules/walker.py

```diff
@@ -0,0 +1,94 @@
+        'walkername': 'name'
+    }
+    configuration = {
+      "fields": {
+            "walkername": {
+                "settings": {
+                    "type": "text",
+                    "require": True,
+                    "description": "Name of walker",
+                    "expected": str
+                }
+            },
+            "setup": {
+                "settings": {
+                    "type": "list",
+                    "require": True,
+                    "empty": [],
+                    "description": "Order of areas",
+                    "expected": list,
+                    "uri": True,
+                    "data_source": "walkerarea",
+                    "uri_source": "api_walkerarea"
+                }
+            }
+        }
+    }
+
+    def get_dependencies(self):
+        sql = 'SELECT `device_id` FROM `settings_device` WHERE `walker_id` = %s'
+        dependencies = self._dbc.autofetch_column(sql, args=(self.identifier,))
+        for ind, device_id in enumerate(dependencies[:]):
+            dependencies[ind] = ('device', device_id)
+        return dependencies
+
+    def delete(self):
+        # Get all walkerareas to determine if they need to be cleaned up
+        sql = "SELECT `walkerarea_id` FROM `settings_walker_to_walkerarea` WHERE `walker_id` = %s"
+        walkerareas = self._dbc.autofetch_column(sql, args=(self.identifier,))
+        super().delete()
+        # Walkerarea cleanup if they are no longer aligned to anything
+        sql = "SELECT COUNT(*) FROM `settings_walker_to_walkerarea` WHERE `walkerarea_id` = %s"
+        for walkerarea_id in walkerareas:
+            in_use = self._dbc.autofetch_value(sql, args=(walkerarea_id,))
+            if not in_use:
+                walkerarea = WalkerArea(self._logger, self._data_manager, identifier=walkerarea_id)
+                walkerarea.delete()
+
+    def _load(self):
+        super()._load()
+        mon_query = "SELECT `walkerarea_id`\n"\
+                    "FROM `settings_walker_to_walkerarea`\n"\
+                    "WHERE `walker_id` = %s ORDER BY `area_order` ASC"
+        mons = self._dbc.autofetch_column(mon_query, args=(self.identifier))
+        self._data['fields']['setup'] = mons
+
+    def save(self, force_insert=False):
+        self.presave_validation()
+        core_data = {
+            'walkername': self._data['fields']['walkername']
+        }
+        super().save(core_data=core_data, force_insert=force_insert)
+        # Get all current walkerareas
+        sql = "SELECT `walkerarea_id` FROM `settings_walker_to_walkerarea` WHERE `walker_id` = %s"
+        walkerareas = self._dbc.autofetch_column(sql, args=(self.identifier,))
+        removed_walkerareas = set(walkerareas) - set(self._data['fields']['setup'])
+        # Remove old walkerareas from the table
+        del_data = {
+            'walker_id': self.identifier
+        }
+        self._dbc.autoexec_delete('settings_walker_to_walkerarea', del_data)

```

Comment:
This is a potential threat to data safety. If anything happens between deletion and insertion, the configuration might be destroyed. 

The easiest way to fix this is to wrap the `save` methods in a database transaction, commit on success and rollback on exception. This should not happen within each `save` method (as they might be nested). Instead, create a `save` method on the base class to do what all model classes have in common (transaction, presave validation etc), and an abstract `_save` which does the per-model work.

Reply:
Ill need to do some extra work to allow transactions as the current implementation of execute is only good for a single query.  I think using __enter__ and __exit__ would be a good way to retrieve a cursor for multiple queries (start transaction, queries, commit).